### PR TITLE
proxy: prepare intercept mode for dynamic channels

### PR DESCRIPTION
**Pull request by hardening**
This patch introduce the basic architecture to intercept content of dynamic channel. When a dynamic channel is in intercept mode, we accumulate and reassemble the current packet so that it can be passed for inspection.

### CODE REVIEW

File: server/proxy/channels/pf_channel_drdynvc.c [hunk 1]
```diff
@@ -93,16 +107,27 @@ static pServerDynamicChannelContext* DynamicChannelContext_new(pServerContext* p
...
+	if (!c)
+		return;
+
+	if (c->backTracker.currentPacket)
+		Stream_Free(c->backTracker.currentPacket, TRUE);
+
+	if (c->frontTracker.currentPacket)
+		Stream_Free(c->frontTracker.currentPacket, TRUE);
+
+	if (c->channelDataDtor)
```
Comment (akallabeth) [inline server/proxy/channels/pf_channel_drdynvc.c:126]: `IFCALL` ?

**Review by akallabeth — APPROVED** [review]
LGTM

File: server/proxy/channels/pf_channel_drdynvc.c (hunk 1 above)
Comment (hardening) [reply server/proxy/channels/pf_channel_drdynvc.c:126]: with  ` IFCALL`  arg1 is the callback to test and to run ? Here we're testing `c->frontTracker.currentPacket` and run `Stream_Free`

File: server/proxy/channels/pf_channel_drdynvc.c [hunk 2]
```diff
@@ -444,27 +498,47 @@ static PfChannelResult DynvcTrackerPeekFn(ChannelStateTracker* tracker, BOOL fir
...
-			WLog_DBG(TAG, "TODO: implement intercepted dynamic channel");
-			return PF_CHANNEL_RESULT_DROP;
+			if (trackerState->dataCallback)
+			{
+				result = trackerState->dataCallback(pdata->ps, dynChannel, isBackData, tracker,
+				                                    firstPacket, lastPacket);
+			}
+			else
+			{
+				WLog_ERR(TAG, "not intercept callback for channel %s(fromBack=%d), dropping packet",
```
Comment (hardening) [inline server/proxy/channels/pf_channel_drdynvc.c:519]: arf typo, do not merge !

**Pull request merged**